contexts/theme-provider.tsx

```diff
@@ -0,0 +1,32 @@
+'use client'
+
+import { usePathname } from "next/navigation";
+import React, { useEffect } from "react";
+
+
+function ThemeProvider({children}: {children: React.ReactNode}) {
+    const pathname = usePathname();
+
+    useEffect(() => {
+        const style = getComputedStyle(document.body);
+
+        if (pathname.includes("arbovirus")) {
+          document.documentElement.style.setProperty('--background', style.getPropertyValue('--arbo'));
+          document.documentElement.style.setProperty('--background-hover',  style.getPropertyValue('--arbo-hover'));
+        } else if (pathname.includes("sarscov2")) {
+          document.documentElement.style.setProperty('--background', style.getPropertyValue('--sc2'));
+          document.documentElement.style.setProperty('--background-hover', style.getPropertyValue('--sc2-hover'));
+        } else {
+          document.documentElement.style.setProperty('--background', style.getPropertyValue('--sero'));
+          document.documentElement.style.setProperty('--background-hover', style.getPropertyValue('--sero-hover'));
+        }
+      }, [pathname]);
+    
+    return (
+        <div>
```

Comment:
Honestly not 100% sure if there is a better way of doing this but this works p well rn and I do not see any problems with it

Reply:
Yeah totally fine imo :)